### PR TITLE
Rename ILockStatement.LockedObject to Expression.

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/21288. Tagging @dotnet/analyzer-ioperation @dotnet/roslyn-compiler for review.

### CODE REVIEW

**Review by genlu — APPROVED** [review]
Consider adding a few tests since you are touch `ILockStatement` interface already :)

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 1]
```diff
@@ -2919,7 +2919,7 @@ public LazyLockStatement(Lazy<IOperation> lockedObject, Lazy<IOperation> body, S
             _lazyBody = body ?? throw new System.ArgumentNullException(nameof(body));
         }
 
-        protected override IOperation LockedObjectImpl => _lazyLockedObject.Value;
+        protected override IOperation ExpressionImpl => _lazyLockedObject.Value;
```
Comment (cston) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: `_lazyExpression.Value`?

**Review by cston — APPROVED** [review]

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 2]
```diff
@@ -286,7 +286,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
         End Function
 
         Private Function CreateBoundMeReferenceOperation(boundMeReference As BoundMeReference) As IInstanceReferenceExpression
-            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, InstanceReferenceKind.Implicit, InstanceReferenceKind.Explicit)
+            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, instanceReferenceKind.Implicit, instanceReferenceKind.Explicit)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: This looks like an unintentional change in casing.

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >instanceReferenceKind [](start = 139, length = 21)

Same here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 3]
```diff
@@ -1073,7 +1073,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundGotoStatementOperation(boundGotoStatement As BoundGotoStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundGotoStatement.Label
-            Dim branchKind As BranchKind = BranchKind.GoTo
+            Dim branchKind As BranchKind = branchKind.GoTo
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Same here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 4]
```diff
@@ -1083,7 +1083,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundContinueStatementOperation(boundContinueStatement As BoundContinueStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundContinueStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Continue
+            Dim branchKind As BranchKind = branchKind.Continue
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Same here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 5]
```diff
@@ -1093,7 +1093,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundExitStatementOperation(boundExitStatement As BoundExitStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundExitStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Break
+            Dim branchKind As BranchKind = branchKind.Break
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Same here.

**Comment by AlekseyTs** [comment]
It looks like we have no tests to verify the change.

**Comment by AlekseyTs** [comment]
Done with review pass.

**Comment by 333fred** [comment]
@AlekseyTs addressed feedback.

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by AlekseyTs — APPROVED** [review]
LGTM

**Comment by 333fred** [comment]
@dotnet-bot retest ubuntu_14_debug_prtest please.

**Comment by 333fred** [comment]
Appears to be a nuget cache corruption issue:
```
10:40:12 /mnt/j/workspace/dotnet_roslyn/features_ioperation/ubuntu_14_debug_prtest/Binaries/dotnet-cli/sdk/1.0.1/Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3106: Assembly strong name "/home/dotnet-bot/.nuget/packages/system.runtime.loader/4.3.0/ref/netstandard1.5/System.Runtime.Loader.dll" is either a path which could not be found or it is a full assembly name which is badly formed. If it is a full assembly name it may contain characters that need to be escaped with backslash(\). Those characters are Equals(=), Comma(,), Quote("), Apostrophe('), Backslash(\). [/mnt/j/workspace/dotnet_roslyn/features_ioperation/ubuntu_14_debug_prtest/src/Scripting/Core/Scripting.csproj]
10:40:12 /mnt/j/workspace/dotnet_roslyn/features_ioperation/ubuntu_14_debug_prtest/Binaries/dotnet-cli/sdk/1.0.1/Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3106: Assembly strong name "/home/dotnet-bot/.nuget/packages/system.runtime.loader/4.3.0/ref/netstandard1.5/System.Runtime.Loader.dll" is either a path which could not be found or it is a full assembly name which is badly formed. If it is a full assembly name it may contain characters that need to be escaped with backslash(\). Those characters are Equals(=), Comma(,), Quote("), Apostrophe('), Backslash(\). [/mnt/j/workspace/dotnet_roslyn/features_ioperation/ubuntu_14_debug_prtest/src/Scripting/Core/Scripting.csproj]
10:40:15 
Hosting/AssemblyLoader/CoreAssemblyLoaderImpl.cs(7,22): error CS0234: The type or namespace name 'Loader' does not exist in the namespace 'System.Runtime' (are you missing an assembly reference?) [/mnt/j/workspace/dotnet_roslyn/features_ioperation/ubuntu_14_debug_prtest/src/Scripting/Core/Scripting.csproj]
```
https://ci.dot.net/job/dotnet_roslyn/job/features_ioperation/job/ubuntu_14_debug_prtest/484/console

**Pull request merged**